gc/base/MemoryPoolAddressOrderedList.cpp

```diff
@@ -634,9 +683,29 @@ MM_MemoryPoolAddressOrderedList::internalAllocateTLH(MM_EnvironmentBase *env, ui
 	consumedSize = (maximumSizeInBytesRequired > freeEntrySize) ? freeEntrySize : maximumSizeInBytesRequired;
 	_largeObjectAllocateStats->decrementFreeEntrySizeClassStats(freeEntrySize);
 
-	/* If the leftover chunk is smaller than the minimum size, hand it out */
+	if (!alignTLHForGC(env, freeEntry, &consumedSize)) {
+		/* If alignment was required and it failed (i.e resulted in consumedSize < minEntrySize), abandon entry and retry */
+		abandonHeapChunk((void *)freeEntry, (void *)(((uint8_t *)freeEntry) + freeEntrySize));
```

Comment:
Should we not look at the leftover (recycleEntrySize  = freeEntrySize - consumedSize, later in the code) and consider 2 cases?
If recycleEntrySize < _minimumFreeEntrySize, then we indeed abandon the whole entry
else we recycle consumedSize part of that entry, bump up freeEntry by consumedSize and try to allocate from there

Reply:
Added TODO for this